Add customizable Standard wish pools and 10-pull costs

SvardXI opened the pull request:
## Description
Added the ability for users to change what items they want on the Standard wish for each of their banners, as well as the ability to to adjust the cost of 10-pulls for discounted banners.

Note: If the field(s) in the JSON is empty for the Standard pools, the server will still use mHY's official Standard pool. Additionally, while the cost of a 10-pull will show in the game's UI, it will only do it before making a 10-pull. The game will still check for (and consume) the adjusted cost.

## Type of changes

<!--- Put an `x` in all the boxes that apply your changes. -->

- [ ] Bug fix
- [ ] New feature 
- [x] Enhancement
- [ ] Documentation

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] My pull request is unique and no other pull requests have been opened for these changes
- [x] I have read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md)
- [x] I am responsible for any copyright issues with my code if it occurs in the future.

SourPatchKidd committed:
Allow customizable Standard wish pools and 10-pull cost

Allow for users to change the 10-pull cost, and to customize Standard wish pulls on banners.

Note: While the discount WILL show up in-game, it will only do so before making a wish. It will revert to showing a flat 10-cost afterward, but will still only require/consume the desired amount for a 10-pull.

SourPatchKidd committed:
Allow customizable Standard wish pools and 10-pull cost 

Allow for users to change the 10-pull cost, and to customize Standard wish pulls on banners. If the array in the user's JSON is empty, the server will default to using mHY's official Standard pool.

Note: While the discount WILL show up in-game, it will only do so before making a wish. It will revert to showing a flat 10-cost afterward, but will still only require/consume the desired amount for a 10-pull.

SourPatchKidd committed:
Insert customizable fields

Small addition - just inserting some fields that have existed for a while into the JSON. Also added fields for custom Standard pools, and set the Weapon Banner weight to mHY official.

SvardXI commented:
Recently noticed one issue with my commit--

I used a source file that was a about a day old. As such, line 129 needs to be replaced:
	- String record = "https://"
	+ String record = "http" + (Grasscutter.getConfig().getDispatchOptions().FrontHTTPS ? "s" : "") + "://"

HotaruYS commented on src/main/java/emu/grasscutter/game/gacha/GachaBanner.java:
```diff
@@ -96,7 +126,7 @@ public GachaInfo toProto() {
 	}
 	public GachaInfo toProto(String sessionKey) {
-		String record = "http" + (Grasscutter.getConfig().getDispatchOptions().FrontHTTPS ? "s" : "") + "://"
+		String record = "https://"
```
Please do not revert made in other PRs changed arbitrarily. Ensure your fork is up-to-date or be very careful when merging conflicts.

HotaruYS replied:
https://github.com/Grasscutters/Grasscutter/pull/649

HotaruYS commented:
Conflicts with changes in https://github.com/Grasscutters/Grasscutter/pull/639

---

>I used a source file that was a about a day old. As such, line 129 needs to be replaced:

You will need to do it yourself, otherwise someone would have to make another PR to fix it.

SourPatchKidd committed:
Fixed issue I created with the gacha record

Reverted a line that I mistakenly removed by failing to check that the class was up-to-date upon submitting my pull request.

Melledy commented:
Thank you for wanting to contribute to our project, but i will be closing this PR as these changes were added in #639
